### ISSUE TITLE
first line in Lua script starting with # causes error "expected statement near '#'"

### ISSUE THREAD

**Issue by okroth** [mpeterv/luacheck#211]
Lua allows at the top of a Lua script a special line starting with # to support the shebang mechanism of Unix shells.
The Lua interpreter skips such a line.
I could not find an option to let luacheck do the same

**Comment by okroth** [comment]
I found that in lexer.lua, line 690, there IS a test for a shebang line, but it requires the line to begin with #! , where the Lua interpreter just requires the line to begin with # .

**Comment by linuxmaniac** [comment]
please read #198

**Comment by alerque** [comment]
I think this may be a case where Lua is wrong and we can ignore what it might have allowed. As far as I know the so called hashbang / shebang must actually be both characters to valid for the use case of specifying a script interpreter. Are there really any other valid use case that shouldn't throw a warning?

In any event if you still think this is an issue we should address, please open [a new issue](https://github.com/lunarmodules/luacheck/issues/new?body=Migrated+issue+from+https://github.com/mpeterv/luacheck/issues/211) on the [new official repository](https://github.com/lunarmodules/luacheck/). As the previous commenter noted the maintainer of this one passed away and we have no way to resolve issues here any more. It would be appreciated if you close this to help us track what is dealt with and (if applicable) open a new issue on the active repository. Thanks!

**Comment by okroth** [comment]
We have a different use case, where the first line contains a digital signature of the rest of the Lua script, which is hidden from the Lua interpreter by the #-character.
In the meantime we simply modified the lexer.lua in our copy to look only for the '#' and not the '!'

**Comment by alerque** [comment]
I think a bit more orthodox  way to do this would be to use a Lua comment, somewhat like `luacheck` does to pass cues to itself, something like:

```lua
-- signature: <hash>
print("rest of script")
```

Overloading the idea of a hashbang for this seems like an edge case that probably *shouldn't* be allowed. That Lua handles it is probably a mistake or lazy parsing shortcut and you might not want to rely on it.

Of course it is open source and you can do what you like, but as a the current maintainer I'm not convinced this is something we should accommodate in the regular releases.

**Comment by okroth** [comment]
retrospectively, the comment would have been the better choice, but I can't change it any more.
